addons/binding/org.openhab.binding.solarlog/src/main/java/org/openhab/binding/solarlog/handler/SolarLogHandler.java

```diff
@@ -0,0 +1,176 @@
+                SolarLogBindingConstants.CHANNEL_YIELDMONTH, SolarLogBindingConstants.CHANNEL_TYPE_YIELDMONTH));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_YIELDYEAR,
+                SolarLogBindingConstants.CHANNEL_YIELDYEAR, SolarLogBindingConstants.CHANNEL_TYPE_YIELDYEAR));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_YIELDTOTAL,
+                SolarLogBindingConstants.CHANNEL_YIELDTOTAL, SolarLogBindingConstants.CHANNEL_TYPE_YIELDTOTAL));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSPAC,
+                SolarLogBindingConstants.CHANNEL_CONSPAC, SolarLogBindingConstants.CHANNEL_TYPE_CONSPAC));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDDAY,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDDAY, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDDAY));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDYESTERDAY,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDYESTERDAY,
+                SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDYESTERDAY));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDMONTH,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDMONTH, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDMONTH));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDYEAR,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDYEAR, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDYEAR));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_CONSYIELDTOTAL,
+                SolarLogBindingConstants.CHANNEL_CONSYIELDTOTAL, SolarLogBindingConstants.CHANNEL_TYPE_CONSYIELDTOTAL));
+        channelConfigs.add(new SolarLogChannelConfig(SolarLogBindingConstants.CHANNEL_ID_TOTALPOWER,
+                SolarLogBindingConstants.CHANNEL_TOTALPOWER, SolarLogBindingConstants.CHANNEL_TYPE_TOTALPOWER));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Read only
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SolarLogSourceConfig.class);
+        scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    refresh();
+                    updateStatus(ThingStatus.ONLINE);
+                } catch (Exception e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            e.getClass().getName() + ":" + e.getMessage());
+                    logger.debug("Error refreshing source " + getThing().getUID(), e);
+                }
+            }
+
+        }, 0, config.refreshInterval, TimeUnit.SECONDS);
```

Comment:
In your documentation you say that an interval below 15 seconds is not supported, but there is no check to set it to 15 seconds if the user provides a lower value.